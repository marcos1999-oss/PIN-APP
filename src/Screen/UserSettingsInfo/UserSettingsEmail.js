import React from "react";
import { Text, View, SafeAreaView, Dimensions } from "react-native";
import { Formik } from "formik";
import { get } from 'lodash';
import CustomIcon from "../../CustomIcon";
import { TextInput } from "../../Components/Common";
import Image from "react-native-auto-height-image";
import { city2 } from "../../Assets";
import { FormContainer } from "./Common";

/*Style*/
import style from "./style.js";

export default class UserSettingsEmail extends React.Component {
  state = {
    isSubmitting: false,
  };

  form = {
    email: this.props.navigation.getParam('me').email,
  };

  componentDidMount() {
    this.props.navigation.setParams({ submit: this.submit });
  }

  updateSavingStatus({ isSubmitting }) {
    this.setState({ isSubmitting });
    this.props.navigation.setParams({ isSubmitting });
  }

  submit = () => {
    if (this.formRef) {
      this.formRef.handleSubmit();
    }
  };

  onSubmit = (values, actions) => {
    if (this.state.isSubmitting) return;

    this.updateSavingStatus({ isSubmitting: true });

    const params = {
      email: values['email'],
    };

    this.props.navigation.getParam('authActions').updateAccount({
      userType: 'user',
      params,
      onSuccess: () => {
        this.updateSavingStatus({ isSubmitting: false });
        this.props.navigation.goBack();
      },
      onFail: (error) => {
        actions.setFieldError('email', get(error, 'response.data.errors.full_messages.0'));
        this.updateSavingStatus({ isSubmitting: false });
      }
    });
  };

  render() {
    return (
      <SafeAreaView style={style.l_container}>
        <View style={{ alignItems: "center", marginBottom: 15 }}>
          <CustomIcon name="setting" size={40} color={"#fff"} />
        </View>

        <Formik onSubmit={this.onSubmit} initialValues={this.form} ref={(ref) => this.formRef = ref}>
          {(props) => (
            <React.Fragment>
              <FormContainer>
                <Text style={style.formTitle}>Write your email</Text>
                <View style={{ height: 5 }} />
                <View style={{ marginLeft: -5 }}>
                  <TextInput
                    form={props}
                    formKey="email"
                    value={props.values.email}
                    placeholder="example@email.com"
                    placeholderTextColor="#AFAFAF"
                    containerStyle={{ borderColor: "#fff", margin: 0 }}
                    style={{ fontSize: 16 }}
                    setRef={() => {}}
                  />
                  <Text style={style.error}>
                    {props.errors && props.errors.email}
                  </Text>
                </View>
                <View style={{ height: 100 }} />
              </FormContainer>
            </React.Fragment>
          )}
        </Formik>
        <Image style={style.l_bottom} source={city2} width={Dimensions.get("window").width} />
      </SafeAreaView>
    );
  }
};
