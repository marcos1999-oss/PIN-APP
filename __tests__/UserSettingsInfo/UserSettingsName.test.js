/**
 * @format
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import UserSettingsEmail from '../../src/Screen/UserSettingsInfo/UserSettingsEmail';

jest.mock('react-native-cached-images', () => {
  return {
    DocumentDir: () => {},
    ImageCache: {
      get: {
        clear: () => {},
      },
    },
  }
});

test('renders correctly', () => {
  const navigation = {
    getParam: (val) => {return ''},
    setParams: jest.fn(),
  };
  const tree = renderer.create(<UserSettingsEmail navigation={navigation} />).toJSON();
  expect(tree).toMatchSnapshot();
});
