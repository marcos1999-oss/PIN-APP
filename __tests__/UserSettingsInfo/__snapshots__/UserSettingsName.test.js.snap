// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#5F92F3",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 15,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 40,
          },
          undefined,
          Object {
            "fontFamily": "PinAppIcon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      î¤’
    </Text>
  </View>
  <View>
    <View
      style={
        Object {
          "marginTop": -20,
          "paddingHorizontal": 20,
          "zIndex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderBottomWidth": 0.5,
              "borderColor": "#ccc",
              "borderLeftWidth": 0.5,
              "borderRadius": 2,
              "borderRightWidth": 0.5,
              "borderTopWidth": 0.5,
              "elevation": 3,
              "flexWrap": "nowrap",
              "marginBottom": 5,
              "marginLeft": 2,
              "marginRight": 2,
              "marginTop": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 1.5,
            },
            Object {
              "borderRadius": 40,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              },
              Object {
                "borderRadius": 40,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "stretch",
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "paddingHorizontal": 20,
                  "paddingVertical": 20,
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#555555",
                    "fontSize": 14,
                  }
                }
              >
                Write your email
              </Text>
              <View
                style={
                  Object {
                    "height": 5,
                  }
                }
              />
              <View
                style={
                  Object {
                    "marginLeft": -5,
                  }
                }
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderBottomWidth": 1,
                      "borderColor": "#fff",
                      "borderLeftWidth": 0,
                      "borderRadius": undefined,
                      "borderRightWidth": 0,
                      "borderTopWidth": 0,
                      "borderWidth": undefined,
                      "flexDirection": "row",
                      "margin": 0,
                      "marginLeft": 2,
                      "opacity": 1,
                    }
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    autoCorrect={false}
                    editable={true}
                    onChangeText={[Function]}
                    placeholder="example@email.com"
                    placeholderTextColor="#AFAFAF"
                    rejectResponderTermination={true}
                    returnKeyType="next"
                    style={
                      Array [
                        Object {
                          "color": "#000",
                          "flex": 1,
                          "fontSize": 17,
                          "height": 50,
                          "paddingLeft": 5,
                          "paddingRight": 5,
                          "top": 1.5,
                        },
                        Object {
                          "fontSize": 16,
                        },
                      ]
                    }
                    underlineColorAndroid="rgba(0,0,0,0)"
                  />
                </View>
                <Text
                  style={
                    Object {
                      "color": "#FF0000",
                      "fontSize": 12,
                      "paddingHorizontal": 10,
                      "paddingVertical": 10,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "height": 100,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <Image
    onError={[Function]}
    onHeightChange={[Function]}
    source={
      Object {
        "testUri": "../../../assets/img/city2.png",
      }
    }
    style={
      Array [
        Object {
          "height": 0,
          "width": 750,
        },
        Object {
          "bottom": 0,
          "position": "absolute",
        },
      ]
    }
  />
</RCTSafeAreaView>
`;
